resolwe/test_helpers/test_runner.py

```diff
@@ -416,14 +416,18 @@ def run_tests(self, test_labels, **kwargs):
     def _git(self, *args):
         """Helper to run Git command."""
         try:
-            return subprocess.check_output(['git'] + list(args)).strip()
+            return subprocess.check_output(['git'] + list(args)).decode('utf8').strip()
         except subprocess.CalledProcessError:
             raise CommandError("Git command failed.")
 
     @contextlib.contextmanager
     def git_branch(self, branch):
         """Temporarily switch to a different Git branch."""
         current_branch = self._git('rev-parse', '--abbrev-ref', 'HEAD')
+        if current_branch == 'HEAD':
+            # Detached HEAD state, we need to get the actual commit.
+            current_branch = self._git('rev-parse', 'HEAD')
```

Comment:
Looks good!

Also mention this in the CHANGELOG since this impacts Resolwe users that run the test suite where HEAD is in a detached state.